modules/calib3d/src/solvepnp.cpp

```diff
@@ -268,10 +268,10 @@ bool solvePnPRansac(InputArray _opoints, InputArray _ipoints,
     int model_points = 5;
     int ransac_kernel_method = SOLVEPNP_EPNP;
 
-    if( npoints == 4 )
+    if( flags == SOLVEPNP_P3P || flags == SOLVEPNP_AP3P)
     {
         model_points = 4;
-        ransac_kernel_method = SOLVEPNP_P3P;
+        ransac_kernel_method = flags;
```

Comment:
Behavior is changed here.
But previous behavior is not mentioned in documentation (and looks like it is [not tested](http://pullrequest.opencv.org/buildbot/export/opencv_releases/master_coverage-lin64-debug/20170416-220349_10075/coverage_html/opencv/modules/calib3d/src/solvepnp.cpp.gcov.html)), so this may be a legal fix.

Probably in this form:
```
if( flags == SOLVEPNP_P3P || flags == SOLVEPNP_AP3P)
{
    model_points = 4;
    ransac_kernel_method = flags;
}
else if( npoints == 4 )
{
    model_points = 4;
    ransac_kernel_method = SOLVEPNP_P3P;
}
```